osquery/tables/system/windows/certificates.cpp

```diff
@@ -307,6 +311,230 @@ void parseSystemStoreString(LPCWSTR sysStoreW,
+                      certContext->pCertInfo->SerialNumber.cbData),
+      back_inserter(serial));
+  r["serial"] = serial;
+
+  std::string authKeyId;
+  if (certContext->pCertInfo->cExtension != 0) {
+    auto extension = CertFindExtension(szOID_AUTHORITY_KEY_IDENTIFIER2,
+                                       certContext->pCertInfo->cExtension,
+                                       certContext->pCertInfo->rgExtension);
+    if (extension != nullptr) {
+      unsigned long decodedBuffSize = 0;
+      CryptDecodeObjectEx(CERT_ENCODING,
+                          X509_AUTHORITY_KEY_ID2,
+                          extension->Value.pbData,
+                          extension->Value.cbData,
+                          CRYPT_DECODE_NOCOPY_FLAG,
+                          nullptr,
+                          nullptr,
+                          &decodedBuffSize);
+
+      certBuff.resize(decodedBuffSize, 0);
+      std::fill(certBuff.begin(), certBuff.end(), 0);
+      auto decodeRet = CryptDecodeObjectEx(CERT_ENCODING,
+                                           X509_AUTHORITY_KEY_ID2,
+                                           extension->Value.pbData,
+                                           extension->Value.cbData,
+                                           CRYPT_DECODE_NOCOPY_FLAG,
+                                           nullptr,
+                                           certBuff.data(),
+                                           &decodedBuffSize);
+      if (decodeRet != FALSE) {
+        auto authKeyIdBlob =
+            reinterpret_cast<CERT_AUTHORITY_KEY_ID2_INFO*>(certBuff.data());
+
+        boost::algorithm::hex(std::string(authKeyIdBlob->KeyId.pbData,
+                                          authKeyIdBlob->KeyId.pbData +
+                                              authKeyIdBlob->KeyId.cbData),
+                              back_inserter(authKeyId));
+      } else {
+        VLOG(1) << "Failed to decode authority_key_id with (" << GetLastError()
+                << ")";
+      }
+    }
+  }
+  r["authority_key_id"] = authKeyId;
+
+  results.push_back(r);
+}
+
+void findUserPersonalCertsOnDisk(const std::string& username,
+                                 const std::string storeId,
+                                 const std::string sid,
+                                 const std::string storeName,
+                                 const std::string storeLocation,
+                                 QueryData& results) {
+  std::stringstream certsPath;
+  certsPath
+      << "C:\\Users\\" << username
+      << "\\AppData\\Roaming\\Microsoft\\SystemCertificates\\My\\Certificates";
+
+  try {
+    for (fs::directory_entry& x :
+         fs::directory_iterator(fs::path(certsPath.str()))) {
+      std::basic_ifstream<BYTE> inp(x.path().string(), std::ios::binary);
+
+      std::vector<BYTE> encodedCert;
+      auto ret = getEncodedCert(inp, encodedCert);
+      if (!ret.ok()) {
+        return;
```

Comment:
Do you mean to `return` here? Given you're in the `for` loop over the directory iterator wouldn't you want to `continue`?

Reply:
That's a bug, awesome catch 👍